Marlin/Marlin_main.cpp

```diff
@@ -12017,21 +12017,36 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
           #endif
         }
 
-        // Save current position to destination, for use later
-        set_destination_from_current();
-
         #if HAS_LEVELING
           // Set current position to the physical position
           const bool leveling_was_active = planner.leveling_active;
           set_bed_leveling_enabled(false);
         #endif
 
         #if ENABLED(DUAL_X_CARRIAGE)
-
+          if(active_extruder==0)
+          {
+            if(current_position[X_AXIS]< X2_MIN_POS || current_position[X_AXIS]> X2_MAX_POS)
+            {
+              no_move = true;
```

Comment:
I think we should use `tmp_extruder == 1` here instead, since the result is that the X2 items are selected for use.